datalad/core/local/status.py

```diff
@@ -168,6 +168,15 @@ def yield_dataset_status(ds, paths, annexinfo, untracked, recursion_limit,
         eval_submodule_state=eval_submodule_state,
         _cache=cache)
     if annexinfo and hasattr(repo, 'get_content_annexinfo'):
+        if paths:
+            # when an annex query has been requested for specific paths,
+            # exclude untracked files from the annex query (else gh-7032)
+            untracked = [k for k, v in status.items() if
+                         v['state'] == 'untracked']
+            lgr.debug(
+                'Skipping %s.get_content_annexinfo() for untracked paths: %s',
+                repo, paths)
```

Comment:
even though only for `DEBUG` I wonder if it is wise to include all `paths` -- what is there is 1000s of them?  do we ever log all `paths` . I would just log number of them I guess, or may be we have some helper to show like first N and then `...` ?

Reply:
Oh, I agree that this would be sensible. I guess it never occurred to me because the very next line in that piece of code is
```
 lgr.debug('Querying %s.get_content_annexinfo() for paths: %s', repo, paths)
```
which would probably also benefit from such a helper. But I'm not aware of one...